### PR TITLE
Initial workflow configuration

### PR DESCRIPTION
**Pull request by dcramer**
This overhauls workflow notifications and focuses them on participating users, as well as adds additional activity-based emails.

TODO
- [ ] Remove 'regression' default rule (as its now part of workflow)

@getsentry/infrastructure @tkaemming @mattrobenolt 
## Notifications Config

screenshot 2016-07-01 14 36 44
## Subscription Info

screenshot 2016-07-01 14 38 05
## Debug Tools

screenshot 2016-06-30 15 17 50

### CODE REVIEW

**Comment by dcramer** [comment]
This still needs to implement other emails as well as is missing the UI around subscribe/unsubscribe from a single issue, in addition to listing active participants on an issue.

File: src/sentry/plugins/sentry_mail/activity/base.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+            'link': group_link,
+            'activity_link': activity_link,
+        }
+
+    def get_email_type(self):
+        return 'notify.activity.{}'.format(
+            self.activity.get_type_display(),
+        )
+
+    def get_subject(self):
```
Comment (dcramer) [inline src/sentry/plugins/sentry_mail/activity/base.py]: we need to give some thought on what subject should actually be used for these

**Comment by dcramer** [comment]
@ckj will need a bit of your time likely to implement the participants on issue details. I think we can model it off of GitHub, but we should also consider how "seen by" vs "participating" works.

**Comment by ckj** [comment]
@dcramer Okay cool. And yeah, 'seen by' kind of complicates this a bit. I'll give it some thought.

Did you want that work to happen in this branch?

**Comment by dcramer** [comment]
@ckj you could do it on a new branch, since all the code exists in master

**Comment by dcramer** [comment]
Going to also refactor debug previews with this

**Comment by dcramer** [comment]
I overhauled how emails work as well as the debug stuff. Not done but its getting there.

**Comment by dcramer** [comment]
One thing of note, we want to move "New Issue" to these emails, but I'm going to avoid that as part of this changeset since it will need coordinated.

**Comment by dcramer** [comment]
Added a bunch of percy snapshots

**Comment by mitsuhiko** [comment]
Looks good to me. We should probably figure out if we want to change the unsub link in the mails.

**Comment by dcramer** [comment]
One remaining TODO Is to update the list of participants when you sub/unsub

**Pull request merged**